### PR TITLE
FM2-224:Add delete method for Task

### PR DESCRIPTION
**Pull request by sherrif10**
https://issues.openmrs.org/browse/FM2-224
cc @ibacher  @corneliouzbett

### CODE REVIEW

**Comment by corneliouzbett** [comment]
Hello @sherrif10, I have updated your PR. Can you add tests for the changes

**Comment by sherrif10** [comment]
@corneliouzbett  let me add them

File: api/src/main/java/org/openmrs/module/fhir2/providers/util/FhirProviderUtils.java [hunk 1]
```diff
@@ -51,6 +61,15 @@ public static OperationOutcome buildDelete(DomainResource resource) {
 		
 	}
 	
+	@Delete
+	public OperationOutcome deleteTask(@IdParam IdType id) {
+		Task task = taskService.delete(id.getIdPart());
+		if (task == null) {
+			throw new ResourceNotFoundException("Could not find task resource with id " + id.getIdPart() + "to delete");
+		}
+		return FhirProviderUtils.buildDelete(task);
```
Comment (Akayeshmantha) [inline api/src/main/java/org/openmrs/module/fhir2/providers/util/FhirProviderUtils.java:70]: This should go to R3/4 Task Providers. Utils are common methods that can be used by no other services. I am not sure why you thought about adding it here.
look at this PR it is similar to what you work on 

https://github.com/openmrs/openmrs-module-fhir2/pull/211

Comment (corneliouzbett) [reply api/src/main/java/org/openmrs/module/fhir2/providers/util/FhirProviderUtils.java:70]: I had already updated this PR with changes [here](https://github.com/openmrs/openmrs-module-fhir2/commit/2b678a313056508d7cd7bbd73880ce73e64ea956). I guess @sherrif10 force-pushed with the wrong changes/commit. He was only supposed to add tests.

Comment (sherrif10) [reply api/src/main/java/org/openmrs/module/fhir2/providers/util/FhirProviderUtils.java:70]: You may be right however those changes need tests like as you said before

Comment (sherrif10) [reply api/src/main/java/org/openmrs/module/fhir2/providers/util/FhirProviderUtils.java:70]: @corneliouzbett  did you recognise some failing tests

**Comment by sherrif10** [comment]
@corneliouzbett  , Your changes broke the travis ?

**Comment by corneliouzbett** [comment]
Hi @sherrif10!
Those changes shouldn't break the tests. Looking into code changes, the patch is incomplete. Just `cherry-pick` this commit [here](https://github.com/openmrs/openmrs-module-fhir2/commit/55d8dd6a81a5f3c41a05a69dc2555fb974aa54e4) and It will add the necessary code + unit & web tests

**Comment by sherrif10** [comment]
Am closing this ticket for now, will open a new clean PR sorry for that @corneliouzbett

**Pull request closed without merging**